### ISSUE TITLE
Questionable modding instructions in README

### ISSUE THREAD

**Issue by slipher** [Unvanquished/Unvanquished#2715]
1. It is suggested to put all new/changed files for the mod in a new dpkdir which will constitute the package to distribute to players. This only works if you make sure to check out the latest release tag as the base for your mod branch, but no mention is made of this. In practice I think most people want to develop against the latest version of the code. In that case you need to either package the whole unvanquished_src.dpkdir or use Urcheon to get only the changed files.
2. It is suggested to put packages in a `pkg/` co-located with `daemon`. (This suggestion is made in other sections besides modding as well.)  IMO this is a poor workflow because it is frequently necessary to delete the build folder, be it because you switched compilers, a new external_deps package was released, or everything is just mysteriously broken. So I believe non-temporary assets should never be placed in the build folder. I always put released DPKs in the `pkg/` in the default homepath which is much nicer because then they are automatically found from *any* build directory without any extra command line flags.

**Comment by illwieckz** [comment]
> It is suggested to put all new/changed files for the mod in a new dpkdir

Yes, the current readme derivates from a previous one that said that, this part was only extended and maybe reworded but the idea was kept, this can be changed, especially since we imported the submodule back and the rest of the toolchain is maturing.
